pkg/domain/infra/abi/containers.go

```diff
@@ -261,6 +261,24 @@ func (ic *ContainerEngine) ContainerRestart(ctx context.Context, namesOrIds []st
 	return reports, nil
 }
 
+func (ic *ContainerEngine) removeContainer(ctx context.Context, ctr *libpod.Container, options entities.RmOptions) error {
+	err := ic.Libpod.RemoveContainer(ctx, ctr, options.Force, options.Volumes)
+	if err == nil {
+		return nil
+	}
+	logrus.Debugf("Failed to remove container %s: %s", ctr.ID(), err.Error())
+	switch errors.Cause(err) {
+	case define.ErrNoSuchCtr:
+		if options.Ignore {
+			logrus.Debugf("Ignoring error (--allow-missing): %v", err)
+			return nil
+		}
+	case define.ErrCtrRemoved:
+		return nil
```

Comment:
if the container is already removed should we log to user/return an error instead of nil?

Reply:
If I ask for a container to be removed and it is removed, then do I care.  Let's not add noice, when there is a potential race condition.  If we want to log this, then do it at the Debug level.

Reply:
I think we're cool as it, mostly because removal is racy; e.g., with the cleanup process.